packages/state-transition/src/util/epochShuffling.ts

```diff
@@ -109,9 +110,46 @@ export function computeEpochShuffling(
 ): EpochShuffling {
   const activeValidatorCount = activeIndices.length;
 
-  const shuffling = activeIndices.slice();
   const seed = getSeed(state, epoch, DOMAIN_BEACON_ATTESTER);
-  unshuffleList(shuffling, seed);
+  const shuffling = unshuffleList(activeIndices, seed, SHUFFLE_ROUND_COUNT);
+
+  const committeesPerSlot = computeCommitteeCount(activeValidatorCount);
+
+  const committeeCount = committeesPerSlot * SLOTS_PER_EPOCH;
+
+  const committees: Uint32Array[][] = [];
+  for (let slot = 0; slot < SLOTS_PER_EPOCH; slot++) {
+    const slotCommittees: Uint32Array[] = [];
+    for (let committeeIndex = 0; committeeIndex < committeesPerSlot; committeeIndex++) {
+      const index = slot * committeesPerSlot + committeeIndex;
+      const startOffset = Math.floor((activeValidatorCount * index) / committeeCount);
+      const endOffset = Math.floor((activeValidatorCount * (index + 1)) / committeeCount);
+      if (!(startOffset <= endOffset)) {
+        throw new Error(`Invalid offsets: start ${startOffset} must be less than or equal end ${endOffset}`);
+      }
+      slotCommittees.push(shuffling.subarray(startOffset, endOffset));
+    }
+    committees.push(slotCommittees);
+  }
+
+  return {
+    epoch,
+    activeIndices,
+    shuffling,
+    committees,
+    committeesPerSlot,
+  };
+}
+
+export async function computeEpochShufflingAsync(
```

Comment:
add unit tests with random inputs to make sure `unshuffleList()` is deeply equal to `await asyncUnshuffleList()`